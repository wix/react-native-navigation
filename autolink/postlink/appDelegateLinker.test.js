import fs from 'fs';

/**
 * Mocks
 */

jest.mock('../postlink/log', () => ({
  log: console.log,
  logn: console.log,
  warn: console.log,
  warnn: console.log,
  info: console.log,
  infon: console.log,
  debug: console.log,
  debugn: console.log,
  errorn: console.log,
}));

/**
 * Tests
 */

describe('appDelegateLinker', () => {
  beforeEach(() => {});

  it('should work for RN 0.68', () => {
    jest.mock('../postlink/path', () => {
      const { copyFileSync } = require('fs');
      const { tmpdir } = require('os');
      const path = require('path');

      const tmpAppDelegatePath = path.resolve(tmpdir(), 'rnn-tests_AppDelegate.mm');

      copyFileSync(
        path.resolve('autolink/fixtures/rn68/AppDelegate.mm.template'),
        tmpAppDelegatePath
      );

      return {
        appDelegate: tmpAppDelegatePath,
      };
    });

    const AppDelegateLinker = require('./appDelegateLinker');
    const linker = new AppDelegateLinker();

    linker.link();
    const appDelegateContent = fs.readFileSync(linker.appDelegatePath, 'utf8');
    expect(appDelegateContent).toMatchSnapshot();
  });

  it('should work for RN 0.69', () => {
    jest.mock('../postlink/path', () => {
      const { copyFileSync } = require('fs');
      const { tmpdir } = require('os');
      const path = require('path');

      const tmpAppDelegatePath = path.resolve(tmpdir(), 'rnn-tests_AppDelegate.mm');

      copyFileSync(
        path.resolve('autolink/fixtures/rn69/AppDelegate.mm.template'),
        tmpAppDelegatePath
      );

      return {
        appDelegate: tmpAppDelegatePath,
      };
    });

    const AppDelegateLinker = require('./appDelegateLinker');
    const linker = new AppDelegateLinker();

    linker.link();
    const appDelegateContent = fs.readFileSync(linker.appDelegatePath, 'utf8');
    expect(appDelegateContent).toMatchSnapshot();
  });
});
