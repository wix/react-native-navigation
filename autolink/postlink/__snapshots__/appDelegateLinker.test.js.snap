// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appDelegateLinker should work for RN 0.77 with Objective-C 1`] = `
"#import "AppDelegate.h"
#import <ReactNativeNavigation/ReactNativeNavigation.h>

#import <React/RCTBundleURLProvider.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  
  return [super application:application didFinishLaunchingWithOptions:launchOptions];
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

/// This method controls whether the \`concurrentRoot\`feature of React18 is turned on or off.
///
/// @see: https://reactjs.org/blog/2022/03/29/react-v18.html
/// @note: This requires to be rendering on Fabric (i.e. on the New Architecture).
/// @return: \`true\` if the \`concurrentRoot\` feature is enabled. Otherwise, it returns \`false\`.
- (BOOL)concurrentRootEnabled
{
  return true;
}

@end "
`;

exports[`appDelegateLinker should work for RN 0.77 with Swift 1`] = `
"import UIKit
import React
import ReactNativeNavigation
import ReactAppDependencyProvider

@main
class AppDelegate: RNNAppDelegate {
  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
    self.moduleName = "app"
    self.dependencyProvider = RCTAppDependencyProvider()

    // You can add your custom initial props in the dictionary below.
    // They will be passed down to the ViewController used by React Native.
    self.initialProps = [:]

    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }

  override func sourceURL(for bridge: RCTBridge) -> URL? {
    self.bundleURL()
  }

  override func bundleURL() -> URL? {
#if DEBUG
    RCTBundleURLProvider.sharedSettings().jsBundleURL(forBundleRoot: "index")
#else
    Bundle.main.url(forResource: "main", withExtension: "jsbundle")
#endif
  }
} "
`;
